modal/cli/run.py

```diff
@@ -423,9 +423,7 @@ def shell(
     """
     env = ensure_env(env)
 
-    console = Console()
-    if not console.is_terminal:
-        raise click.UsageError("`modal shell` can only be run from a terminal.")
+    pty = sys.stdout.isatty()
```

Comment:
If we want to support this from a non-interactive terminal, we should change the doc comment to reflect this - right now, it only talks about using `shell` for interactive terminals.

Reply:
<img width="1381" alt="image" src="https://github.com/user-attachments/assets/d3195140-82e5-4378-ab88-5929052f750c">
wow this doesn't show up well right now